### PR TITLE
Upgrade dependencies including new bootstrap conventions

### LINKED ISSUE #75: Extremely high CPU use by ffmpeg
MPEG4 to WebM transcoding is extremely slow.  https://farm.openzim.org/pipeline/c06a8148-7d9a-422c-b5b4-abfe93d51168 has been crawling along for two weeks while using 100% of all CPUs.

### LINKED ISSUE #83: The scraper is forking too many FFmpeg instances
It looks like the `--threads` parameter is not doing its intended job, or at least the scraper is starting way too many threads.

At least one of our worker contributor is reporting that `khan_academy` jobs are using 400 encoding threads on its machine because:
- Docker detects 20 hardware threads on its machine
- Kolibri hence forks 20 FFmpeg instance
- Each FFmpeg instance sees that there are 20 hardware threads and spawns 20 of its own threads
- 💥400 threads in parallel ...

### PR DESCRIPTION
**Pull request by benoit74**
Fix #83 
Fix #75 
(both are fixed by upstream change in python-scraperlib 3.3.0)

## Changes

- Upgrade all dependencies before release, include latest Python bootstrap conventions
  - Main change is the adoption of `hatch-openzim` plugin, including the build feature to install JS dependencies
    - I had to drop the special `$SEDEXT` for Darwin machines in sed command for perseus patching (looks like a dev feature anyway) ; I don't think it deserves a feature in `hatch-openzim`
  - Upgrade from Python 3.11 to Python 3.12
  - JS dependencies have not been updated, too complex + some (e.g. bootstrap) will be dropped with new UI (`1.3.0` hopefully)
- Obviously the upgrade to Python scraperlib 3.3.0 is not negligeable (fix two issues, encoder preset change for low webm quality)

## Tests

A test ZIM has been made with following command (`OPTIM_CACHE` points to `org-kiwix-dev-kolibri`)

```
docker run --rm -it -v $PWD/output:/output local-kolibri2zim kolibri2zim --name "minimal_test" --title "Minimal Kolibri Channel Test" --description "This is a minimal Kolibri Channel" --channel-id "7f744ce8d28b471eaf663abd60c92267" --debug --low-quality --use-webm --optimization-cache $OPTIM_CACHE
```

### CODE REVIEW

**Comment by benoit74** [comment]
Build is failing until hatch-openzim 0.2.0 is released (https://github.com/openzim/hatch-openzim/issues/11 issue)

File: get_js_deps.sh [hunk 1]
```diff
@@ -1,110 +0,0 @@
...
-
-echo "getting perseus renderer"
-curl -L -O https://github.com/rgaudin/standalone-perseus/archive/refs/tags/v1.1.4.zip
-rm -rf $ASSETS_PATH/perseus
-unzip -o v1.1.4.zip
-mkdir -p $ASSETS_PATH/perseus
-mv standalone-perseus-1.1.4/* $ASSETS_PATH/perseus
-rm -rf standalone-perseus-1.1.4/
-rm -f v1.1.4.zip
-sed -i $SEDEXT '1s/""/"assets\/perseus\/"/' $ASSETS_PATH/perseus/build/frame-perseus.js
```
Comment (benoit74) [inline get_js_deps.sh:103]: This is where the `$SEDEXT` was

**Comment by benoit74** [comment]
As usual, I forgot to update the CHANGELOG, done now, sorry

**Review by rgaudin — APPROVED** [review]
LGTM ; good to have a reference implementation of the plugin!

**Pull request merged**